src/Loadable.C

```diff
@@ -0,0 +1,78 @@
+#include "Loadable.h"
+
+#include <fstream>
+#include <regex>
+
+namespace paratreet {
+
+parameters load_parameters(const char* file) {
+  parameters map;
+  std::fstream fs(file, std::ios::in);
+  CmiAssert(fs.is_open());
+
+  std::string line;
+  std::regex line_regex(
+      "([_a-zA-Z0-9]+)\\s*=\\s*([^\\s#]+).*");  // matches (id) = (value)
+  while (std::getline(fs, line)) {
+    // skip comments and empty lines
+    if (line.empty() || line[0] == '#') {
+      continue;
+    }
+    std::smatch m;
+    // try to match the line
+    if (std::regex_match(line, m, line_regex)) {
+      // extract key and value if successful
+      auto& key = m[1];
+      auto& val = m[2];
+      // determine the type from the key
+      auto ty = Value::get_type(key);
+      // emplace the value with the appropriate conversion
+      auto ins = ([&](void) {
+        using T = Value::type;
+        switch (ty) {
+          case T::kBool:
+            return map.emplace(key, (bool)std::stoi(val));
```

Comment:
I'm getting a compile error here with gcc v4.8.5 20150623 (Red Hat 4.8.5-44)
Loadable.C:34:57:   required from here
/usr/include/c++/4.8.2/bits/stl_pair.h:134:45: error: use of deleted function ‘paratreet::Value::Value(const paratreet::Value&)’
  : first(std::forward<_U1>(__x)), second(__y) { }
                                             ^
In file included from Field.hpp:4:0,
                 from Loadable.h:4,
                 from Loadable.C:1:
Value.hpp:54:3: error: declared here
   Value(const Value&) = delete;
   ^